run-tests.php

```diff
@@ -3413,303 +3430,281 @@ function show_result(
 
 }
 
-function junit_init(): void
+class JUnit
 {
-    // Check whether a junit log is wanted.
-    global $workerID;
-    $JUNIT = getenv('TEST_PHP_JUNIT');
-    if (empty($JUNIT)) {
-        $GLOBALS['JUNIT'] = false;
-        return;
-    }
-    if ($workerID) {
-        $fp = null;
-    } elseif (!$fp = fopen($JUNIT, 'w')) {
-        error("Failed to open $JUNIT for writing.");
-    }
-    $GLOBALS['JUNIT'] = [
-        'fp' => $fp,
-        'name' => 'PHP',
+    private bool $enabled = true;
+    private $fp = null;
+    private array $suites = [];
+    private array $rootSuite = self::EMPTY_SUITE + ['name' => 'php'];
+
+    private const EMPTY_SUITE = [
         'test_total' => 0,
         'test_pass' => 0,
         'test_fail' => 0,
         'test_error' => 0,
         'test_skip' => 0,
         'test_warn' => 0,
+        'files' => [],
         'execution_time' => 0,
-        'suites' => [],
-        'files' => []
     ];
-}
 
-function junit_save_xml(): void
-{
-    global $JUNIT;
-    if (!junit_enabled()) {
-        return;
+    public function __construct(array $env, int $workerID)
+    {
+        // Check whether a junit log is wanted.
+        $fileName = $env['TEST_PHP_JUNIT'] ?? null;
+        if (empty($fileName)) {
+            $this->enabled = false;
+            return;
+        }
+        if (!$workerID && !$this->fp = fopen($fileName, 'w')) {
```

Comment:
The check for workerID here seems like something of an abstraction leak. I'm also not really clear on how this worked if no FP is initialized. Won't subsequent operations on $this->fp fail?

Reply:
I personally prefer to avoid mixing refactoring with behavior changes in the same commit, to avoid bamboozling the reviewers. Can do otherwise if this reviewer prefers so:) `$this->fp` is only used in saveXML() which never gets called from workers - guess this is another issue to fix in subsequent patches.

Reply:
I see ... I guess the proper way to handle it would be to move saveXML outside this class, and make this one just hold the data.

But you're right, let's leave it as-is for now.